apps/meeting/src/providers/VideoGridProvider/state.tsx

```diff
@@ -0,0 +1,510 @@
+        videoSourceState,
+      };
+    }
+    case VideoGridAction.UpdateActiveSpeakers: {
+      const { activeSpeakers } = payload;
+      const activeSpeakersWithCameraSource = [];
+      const { cameraSources } = videoSourceState;
+
+      for (const attendeeId of activeSpeakers) {
+        if (
+          attendeeStates[attendeeId]?.videoEnabled &&
+          cameraSources.includes(attendeeId)
+        ) {
+          activeSpeakersWithCameraSource.push(attendeeId);
+        }
+      }
+
+      videoSourceState.activeSpeakersWithCameraSource = activeSpeakersWithCameraSource;
+      updateDownlinkPreferences(gridState, videoSourceState, attendeeStates);
+
+      return {
+        ...state,
+        videoSourceState,
+      };
+    }
+    case VideoGridAction.UpdateLocalSourceState: {
+      const {
+        isVideoEnabled,
+        localAttendeeId,
+        isLocalUserSharing,
+        sharingAttendeeId,
+      } = payload;
+
+      videoSourceState.hasLocalVideo = isVideoEnabled;
+      videoSourceState.hasLocalContentSharing = isLocalUserSharing;
+      videoSourceState.contentShareId = sharingAttendeeId;
+
+      if (localAttendeeId && localAttendeeId in attendeeStates) {
+        attendeeStates[localAttendeeId].videoEnabled = isVideoEnabled;
+      }
+
+      updateDownlinkPreferences(gridState, videoSourceState, attendeeStates);
+
+      return {
+        ...state,
+        attendeeStates,
+        videoSourceState,
+      };
+    }
+    case VideoGridAction.UpdateGridState: {
+      const { videoGridMode } = payload;
+      gridState.mode = videoGridMode;
+      updateDownlinkPreferences(gridState, videoSourceState, attendeeStates);
+
+      return {
+        ...state,
+        gridState,
+      };
+    }
+    case VideoGridAction.PauseVideoTile: {
+      const { attendeeId } = payload;
+      attendeeStates[attendeeId].bandwidthConstrained = true;
+
+      return {
+        ...state,
+        attendeeStates,
+      };
+    }
+    case VideoGridAction.UnpauseVideoTile: {
+      const { attendeeId } = payload;
+      attendeeStates[attendeeId].bandwidthConstrained = false;
+
+      return {
+        ...state,
+        attendeeStates,
+      };
+    }
+    case VideoGridAction.ZoomIn: {
+      const { threshold } = gridState;
+      const { cameraSources, hasLocalVideo } = videoSourceState;
+      const numberOfTiles = cameraSources.length + (hasLocalVideo ? 1 : 0);
+
+      if (numberOfTiles > threshold) {
+        gridState.isZoomed = true;
+        updateDownlinkPreferences(gridState, videoSourceState, attendeeStates);
+      }
+
+      return {
+        ...state,
+        gridState,
+      };
+    }
+    case VideoGridAction.ZoomOut: {
+      if (gridState.isZoomed) {
+        gridState.isZoomed = false;
+        updateDownlinkPreferences(gridState, videoSourceState, attendeeStates);
+      }
+
+      return {
+        ...state,
+      };
+    }
+    default:
+      throw new Error('Incorrect type in VideoGridStateProvider');
```

Comment:
The filename says `VideoGridProvider`,  so need to update `VideoGridStateProvider` to the same.

Reply:
Good catch!